ls1intum/Artemis
# Add OCaml as programming language.

### Checklist

### Motivation and Context
One course wants to use OCaml for programming exercises.

### Description
This PR adds OCaml as a programming language. A template is included in the PR and a docker image (https://github.com/ls1intum/artemis-ocaml-docker) is also present.
Currently, OCaml is only supported for Bamboo, not for Jenkins.

The template was provided by @jerhard.

### Steps for Testing
**Note:** As OCaml is only supported for Bamboo, use TS1 or TS3 for testing.
1. Create a programming exercise with OCaml
2. Verify that the solution repository fulfills all (7) test cases and the assignment repository fulfills no test case.
3. Participate in the exercise as a student and solve some/all of the tasks. In case you are not that familiar with OCaml, you can find the solution here: https://github.com/ls1intum/Artemis/blob/feature/ocaml/src/main/resources/templates/ocaml/solution/src/assignment.ml
4. Delete the exercise from the test server as it might cause issues when testing other PRs


### Test Coverage
Unchanged
